File: lists/queue.rb
require_relative 'linked_list'

class Queue
  def initialize
    @list = LinkedList.new
  end

  def peek
    @list.head&.value
  end

  def enqueue(value)
    @list.append(value)
  end

  def dequeue
    result = @list.head&.value
    @list.delete(result)
    result
  end
end

File: lists/queue.py
from linked_list import LinkedList

class Queue(LinkedList):
  def peek(self):
    if self.head():
      return self.head().value

  def enqueue(self, value):
    self.append(value)

  def dequeue(self):
    if self.head():
      result = self.head().value
      self.delete(result)
      return result